> Seems that `go test` is unhappy about `-tags` given more than once. From [the runtime test](https://jenkins.cilium.io/job/Cilium-PR-Runtime-4.9/1313/):
> 
> ```
> Makefile:148: recipe for target 'tests-privileged' failed
> go test: tags flag may be set only once
> ```

FWIW, with the upcoming Go version 1.15, this limitation will be gone, see golang/go@583419e5d2a893e369095355035f4ebb18bb85f8.

Nevertheless, it's of course much nicer to only have `-tags` specified once the way this PR does it.

--- a/Makefile
+++ b/Makefile
@@ -56,8 +56,6 @@ JOB_BASE_NAME ?= cilium_test
 GO_VERSION := $(shell cat GO_VERSION)
 GOARCH := $(shell $(GO) env GOARCH)
 
-comma:= ,
-
 TEST_LDFLAGS=-ldflags "-X github.com/cilium/cilium/pkg/kvstore.consulDummyAddress=https://consul:8443 \
 	-X github.com/cilium/cilium/pkg/kvstore.etcdDummyAddress=http://etcd:4002 \
 	-X github.com/cilium/cilium/pkg/testutils.CiliumRootDir=$(ROOT_DIR) \
@@ -87,11 +85,8 @@ endef
 
 define generate_k8s_api_all
 	$(call generate_k8s_api,all,github.com/cilium/cilium/pkg/k8s/client,$(1),$(2))
-	$(call generate_deepequal,"$(subst $(space),$(comma),$(foreach pkg,$(2),$(1)/$(subst ",,$(subst :,/,$(pkg)))))")
+	$(call generate_deepequal,"$(call join-with-comma,$(foreach pkg,$(2),$(1)/$(subst ",,$(subst :,/,$(pkg)))))")
 endef
-
-empty:=
-space:= $(empty) $(empty)
 
 define generate_k8s_api_deepcopy_deepequal
 	$(call generate_k8s_api,deepcopy,github.com/cilium/cilium/pkg/k8s/client,$(1),$(2))
@@ -102,12 +97,12 @@ define generate_k8s_api_deepcopy_deepequal
 	@#    "$pkg", with the characters replaced, create a new string with the
 	@#    prefix $(1)
 	@#   Finally replace all spaces with commas from the generated strings.
-	$(call generate_deepequal,"$(subst $(space),$(comma),$(foreach pkg,$(2),$(1)/$(subst ",,$(subst :,/,$(pkg)))))")
+	$(call generate_deepequal,"$(call join-with-comma,$(foreach pkg,$(2),$(1)/$(subst ",,$(subst :,/,$(pkg)))))")
 endef
 
 define generate_k8s_api_deepcopy_deepequal_client
 	$(call generate_k8s_api,deepcopy$(comma)client,github.com/cilium/cilium/pkg/k8s/slim/k8s/client,$(1),$(2))
-	$(call generate_deepequal,"$(subst $(space),$(comma),$(foreach pkg,$(2),$(1)/$(subst ",,$(subst :,/,$(pkg)))))")
+	$(call generate_deepequal,"$(call join-with-comma,$(foreach pkg,$(2),$(1)/$(subst ",,$(subst :,/,$(pkg)))))")
 endef
 
 define generate_k8s_protobuf

--- a/Makefile.defs
+++ b/Makefile.defs
@@ -138,7 +138,7 @@ GO_BUILD += $(GO_BUILD_FLAGS)
 GO_BUILD_WITH_CGO += $(GO_BUILD_FLAGS)
 
 GO_TEST = $(GO) test $(GO_TEST_FLAGS)
-GO_CLEAN = $(GO) clean $(GO_TEST_FLAGS)
+GO_CLEAN = $(GO) clean $(GO_CLEAN_FLAGS)
 # TODO: remove `GO111MODULE=off` once Go 1.13 is deprecated by Go maintainers
 GO_VET = GO111MODULE=off $(GO) vet
 GO_LIST = GO111MODULE=off $(GO) list

--- a/Makefile.defs
+++ b/Makefile.defs
@@ -4,6 +4,12 @@
 SHELL := /bin/bash
 .SHELLFLAGS := -eu -o pipefail -c
 
+# define a function replacing spaces with commas in a list
+empty :=
+space := $(empty) $(empty)
+comma := ,
+join-with-comma = $(subst $(space),$(comma),$(strip $1))
+
 ROOT_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 RELATIVE_DIR := $(shell echo $(realpath .) | sed "s;$(ROOT_DIR)[/]*;;")
 include $(ROOT_DIR)/Makefile.quiet

--- a/pkg/api/const.go
+++ b/pkg/api/const.go
@@ -1,4 +1,4 @@
-// Copyright 2017-2018 Authors of Cilium
+// Copyright 2017-2020 Authors of Cilium
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -20,12 +20,10 @@ import (
 )
 
 const (
-	// GroupFilePath is the unix group file path.
-	GroupFilePath = "/etc/group"
 	// CiliumGroupName is the cilium's unix group name.
 	CiliumGroupName = "cilium"
 	// SocketFileMode is the default file mode for the sockets.
 	SocketFileMode os.FileMode = 0660
-	//ClientTimeout specifies timeout to be used by clients
+	// ClientTimeout specifies timeout to be used by clients
 	ClientTimeout = 90 * time.Second
 )

--- a/pkg/api/socket.go
+++ b/pkg/api/socket.go
@@ -1,4 +1,4 @@
-// Copyright 2017-2018 Authors of Cilium
+// Copyright 2017-2020 Authors of Cilium
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -15,12 +15,10 @@
 package api
 
 import (
-	"bufio"
 	"fmt"
-	"io"
 	"os"
+	"os/user"
 	"strconv"
-	"strings"
 
 	"github.com/cilium/cilium/pkg/logging"
 	"github.com/cilium/cilium/pkg/logging/logfields"
@@ -30,34 +28,19 @@ import (
 
 var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "api")
 
-// GetGroupIDByName returns the group ID for the given grpName.
-func GetGroupIDByName(grpName string) (int, error) {
-	f, err := os.Open(GroupFilePath)
+// getGroupIDByName returns the group ID for the given grpName.
+func getGroupIDByName(grpName string) (int, error) {
+	group, err := user.LookupGroup(grpName)
 	if err != nil {
 		return -1, err
 	}
-	defer f.Close()
-	br := bufio.NewReader(f)
-	for {
-		s, err := br.ReadString('\n')
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
-			return -1, err
-		}
-		p := strings.Split(s, ":")
-		if len(p) >= 3 && p[0] == grpName {
-			return strconv.Atoi(p[2])
-		}
-	}
-	return -1, fmt.Errorf("group %q not found", grpName)
+	return strconv.Atoi(group.Gid)
 }
 
-// SetDefaultPermissions sets the given socket to with cilium's default
-// group and mode permissions. Group `CiliumGroupName` and mode `0660`
+// SetDefaultPermissions sets the given socket's group to `CiliumGroupName` and
+// mode to `SocketFileMode`.
 func SetDefaultPermissions(socketPath string) error {
-	gid, err := GetGroupIDByName(CiliumGroupName)
+	gid, err := getGroupIDByName(CiliumGroupName)
 	if err != nil {
 		log.WithError(err).WithFields(logrus.Fields{
 			logfields.Path: socketPath,

